---
title: "P513-4 : Dysregulated asthmatic epithelial interferon responses to viruses drive exacerbation, T2 inflammation, and airway remodeling | Data Cleaning"
output: 
  html_document:
    toc: yes
    toc_float:
      collapsed: no
  pdf_document:
    toc: yes
editor_options:
  chunk_output_type: console
header-includes:
  \usepackage{float}
---

<style type="text/css">
body{ /* Normal  */
      font-size: 14px;
  }
h1 { /* Header 1 */
  font-size: 28px;
}
h2 { /* Header 2 */
    font-size: 24px;
}
h3 { /* Header 3 */
  font-size: 20px;
}
h4 { /* Header 4 */
  font-size: 16px;
}
</style>

```{r include=FALSE}
knitr::opts_chunk$set(fig.height=8.5, fig.width = 8.5, echo=FALSE,message=FALSE, warning=FALSE)
```

# Project Summary

How does the interferon response change post baseline and baseline post rhinovirus infection, and in particular the kinetics, how does it relate to viral clearance, patient history.


The primary questions are: 
How do type 1 and 2 interferon responses on the level of the interferon relate back to host characteristics.


```{r setup, echo=FALSE, message=FALSE, warning=FALSE, cache=FALSE}
library(knitr)
library(tidyverse)

library(ggthemes)
#library(edgeR)
library(RColorBrewer)
library(kableExtra)
library(viridis)
library(ggbeeswarm)
library(gplots)
library(upstartr)
library(janitor)
library(cowplot)
library(edgeR)
library(ggpubr)

library(ComplexHeatmap)
library(limma)
library(data.table)
library(multcomp)

library(bRi)
library(miscHelpers)
library(RNAseQC)
library(countSubsetNorm)
library(apird)


library(randomcoloR)
library(dendextend)
library(corrplot)
library(patchwork)
#library(ggsankey)
# Change package priority to dplyr, not a great way to do it but works for(if it aint broke dont fix it)
select <- dplyr::select
rename <- dplyr::rename
filter <- dplyr::filter
distinct <- dplyr::distinct
mutate <- dplyr::mutate


opts_chunk$set(
  fig.width=6, fig.height=4.25, cache = TRUE,
  echo=FALSE, warning=FALSE, message=FALSE)
options(stringsAsFactors = FALSE)

theme_set(
  theme_bw(20) +
    theme(panel.grid.major = element_blank(),
          panel.grid.minor = element_blank(),
          panel.border = element_rect(colour="black", fill = NA, size = 1),
          axis.text = element_text(colour="black"),
          axis.ticks = element_line(colour="black"),
          axis.text.x = element_text(angle=0)))
update_geom_defaults("point", list(shape=16))
grDevices::pdf.options(useDingbats = FALSE)
options(ggrepel.max.overlaps = Inf)
#Create 'not in' operator
`%notin%` <- Negate(`%in%`)
# test_plot <- ggplot2::ggplot(mapping = ggplot2::aes(x = 1:10, y = 1:10)) + ggplot2::geom_point()
# ggplot(mapping = aes(x = 1:10, y = 1:10)) + geom_point()

col.vec <- c(
  "none" = "#969696",
  "percent aligned" = "#CC79A7",
  "CV coverage, percent aligned" = "steelblue3",
  
  "total seqs, percent aligned" = "#feb24c",
  
  "CV coverage, total seqs, percent aligned" = "#fc4e2a",
  "total seqs" = "darkolivegreen",
  "CV coverage, total seqs" = "aquamarine"
  )

IFN_treatment_col <- c("None" = "grey60", 
                       "CXCL10" = "mediumorchid", 
                       "IFNb" = "darkorange3")

virus_col <- c("None" = "steelblue", 
               "RV16" = "red")

Treatment_col <- c("None + None" = "grey60", 
                   "None + RV16" = "red", 
                   "CXCL10 + RV16" = "mediumpurple4",  # Blend of mediumorchid + firebrick
                   "IFNb + RV16" = "darkred") 
 rasterResolutionDpi <- 300
 
 
 clean_column_names <- function(col_names) {
  col_names %>%
    str_replace_all("\\.\\.\\.+", "_") %>%  # Replace multiple dots with a single underscore
    str_replace_all("\\.", "_") %>%         # Replace remaining dots with underscores
    str_replace_all("_$", "") %>%           # Remove trailing underscores
    str_replace_all("__+", "_") %>%         # Remove double underscores
    str_replace_all("\\s+", "_") %>%        # Replace spaces with underscores
    str_replace_all("[^A-Za-z0-9_]", "")    # Remove any remaining special characters
}
```


```{r setFilenamesAndGlobalVariables}
projectNumber <- "P513-4"
dirBoxBase <- 
  file.path("~", "Library", "CloudStorage", "Box-Box")
dirRoot <-
  file.path(
    dirBoxBase, "Altman_Lab", # Box version
    # "~", "Documents", "Projects_local", # local version
    projectNumber)
dirPlots <- file.path("Figs")
dirresults <- file.path("Results")
dirDataclean <- file.path("Data_clean")
dirDataraw <- file.path("Data_raw")



```

# 1. Load in data

```{r}
#setwd("~/Library/CloudStorage/Box-Box/BRI_Projects/P506-1/")

design <- read_csv("Data_raw/design_raw.csv") %>% arrange(libid) %>% 
  mutate(Treatment = factor(Treatment,c("None + None","None + RV16","CXCL10 + RV16", "IFNb + RV16"))) %>% 
  mutate(treatment = factor(treatment,levels=c("None","CXCL10","IFNb"))) %>% 
  mutate(stimulation = factor(stimulation,levels = c("None","RV16"))) %>% 
  mutate(Exacerbation = factor(Exacerbation,levels=c("No Severe Exacerbation","Severe Exacerbation")))


counts_raw <- read_csv("Data_raw/counts_raw.csv") %>% relocate(c(geneName,design$libid))


```

```{r}
design %>% 
  count(studyGroup,treatment,stimulation,timePoint) %>%

kable(align="l",
      caption="Total libraries") %>% 
  kable_styling(bootstrap_options = "striped", full_width = FALSE)
```

## 2.1 Load key.
## 2.2 Filter gene key to protein coding (pc) genes that occur in the count data set and have valid hgnc symbols. 
```{r key, message=FALSE}

if(file.exists(file.path(dirDataraw,"key_pc.csv"))){
  key.pc <- read_csv("Data_raw/key_pc.csv")
} else {
  library(biomaRt)
  ensembl <- useEnsembl(biomart = "genes", dataset = "hsapiens_gene_ensembl")
  full_key <- biomaRt::getBM(attributes = c("ensembl_gene_id","hgnc_symbol",	
                                "transcript_biotype"), mart = ensembl) %>% 
              #rename variable to match count data
              dplyr::rename(geneName = ensembl_gene_id,gene_biotype = transcript_biotype) 
  
  key.pc <- full_key %>% 
    # Keep only valid hgnc symbols
    filter(!is.na(hgnc_symbol)) %>% 
    # Keep protein coding genes only
    filter(gene_biotype == "protein_coding") %>% 
    # Remove duplicate entries
    distinct(geneName, .keep_all=TRUE) %>% 
    # Keep only genes found in dataset
    filter(geneName %in% counts_raw$geneName) %>% 
    arrange(geneName)
  
  write_csv(key.pc,"Data_raw/key_pc.csv")
}
```


## 2.3 Filter counts to protien coding genes
```{r}
counts.pc <- counts_raw %>% 
  filter(geneName %in% key.pc$geneName) %>% 
  arrange(geneName)
```


# 3. RNAseq Quality Control

## 3.1 Quality control by sequencing and alignment metrics

```{r setQcCuts}
# set QC thresholds
# Usually start with lenient cutoffs and then increase or decrese as needed 
pct_aligned_cutoff = 95
fastq_total_reads_cutoff = 1E6
median_cv_coverage_cutoff = 0.7

```

```{r}
pal.donorId <-
  big_colorblind_pal(
    n_distinct(design$donorId),
    shuffle_colors = TRUE,
    drop_yellow = TRUE, drop_black = TRUE) %>%
  setNames(
    design %>%
      dplyr::arrange(donorId) %>%
      dplyr::pull(donorId) %>%
      unique())

```
# 3.2 Look at Median CV coverage

```{r cv.dups, echo=FALSE}
if (!dir.exists("Figs/cleaning")) dir.create("Figs/cleaning",recursive = TRUE)
median_cv_plot <- design %>% 
ggplot(aes(median_cv_coverage, pct_aligned)) +
           geom_point(aes(color=Treatment),
                      size=3) +
  #Label points outside cutoffs
  geom_text_repel(data=filter(design,
                            median_cv_coverage > median_cv_coverage_cutoff | 
                            pct_aligned < pct_aligned_cutoff),
            aes(label=libid), show.legend = FALSE) +
  # Add cutoff lines
  geom_vline(xintercept=median_cv_coverage_cutoff, linetype=2) +
  geom_hline(yintercept=pct_aligned_cutoff, linetype=2) +
  #Beautify
  theme_classic() +
  theme(legend.position = "none") +
  labs(x="Median CV coverage", y="Percent aligned") + scale_color_manual(values = Treatment_col)
  # lims(x=c(0,1.2), y=c(0,1))

median_cv_plot
#ggsave("Figs/cleaning/median_cv_coverage.pdf",plot = median_cv_plot,width = 15)
ggsave("Figs/cleaning/median_cv_coverage.png",plot = median_cv_plot,height = 8)
```

### 3.3 Plot read counts 

```{r}
design_w_QC <- design %>%  mutate(col.group = ifelse(fastq_total_reads >= 1E6, "1,000,000+",
                            ifelse(fastq_total_reads >= 5E5, "500,000+",
                                   ifelse(fastq_total_reads >= 2.5E5, "250,000+",
                                          ifelse(fastq_total_reads >= 1E5, "100,000+",
                                                 "Less than 100,000")))),
          col.group = factor(col.group, levels=c("1,000,000+","500,000+","250,000+",
                                                 "100,000+","Less than 100,000")))

total_reads_QC_plot <- design_w_QC %>% ggplot(aes(x=reorder_within(libid, by=fastq_total_reads,within = col.group), 
           y=fastq_total_reads, fill=Treatment))  +
    geom_col() +

    # Add cutoff line
    geom_hline(yintercept = fastq_total_reads_cutoff) +

    # Beautify
    theme_classic()  +
    theme(axis.text.x = element_text(size=rel(0.75),
                                     angle = 90, hjust = 1)) +
    labs(x="libID", y="Total reads", fill="") +
    scale_x_reordered() +
    scale_y_continuous(trans = 'log10',
    breaks = trans_breaks("log10", function(x) 10^x),
    labels = trans_format("log10", math_format(10^.x))) + scale_fill_manual(values = Treatment_col)

QC_combined_plot <- median_cv_plot+total_reads_QC_plot+plot_layout(nrow=1,guides = "collect")

ggsave(file.path(dirPlots,paste("cleaning/total_reading_medianCV_comb.pdf",sep = "")),plot = QC_combined_plot,height = 5,width = 12)
```

# 3.3 Plot total counts per library. Libraries outside cutoffs are labeled as "questionable". The minimum total sequences cutoff set above is indicated by a horizontal line.
No samples were QC questionable

# PCA plots for QC
```{r  echo=FALSE, warning=FALSE}
#Calculate PCA for all data.
PCA.all <- counts.pc %>% 
  column_to_rownames("geneName") %>% 
  #Convert to log counts per million
  cpm(., log=TRUE) %>% 
  t() %>% 
  #Calc PCA
  prcomp()

PC1.label <- paste("PC1 (", summary(PCA.all)$importance[2,1]*100, "%)", sep="")
PC2.label <-paste("PC2 (", summary(PCA.all)$importance[2,2]*100, "%)", sep="")

# Extract PC values
PCA.all.dat <- as.data.frame(PCA.all$x) %>% 
  rownames_to_column("libid") %>% 
  # Select PCs for plotting
  dplyr::select(libid, PC1:PC3) %>% 
  # Merge with metadata
  left_join(design_w_QC, by="libid")

plot_pca_unnorm_by_qc <- ggplot(PCA.all.dat, aes(PC1, PC2)) +
           geom_point(aes(color=Treatment),alpha=0.8,
                      size=3) + 
  #Beautify
  theme_classic() +
  theme(legend.title = element_blank(),
        legend.position = "bottom") +
  labs(x=PC1.label, y=PC2.label, title="Un-normalized logCPM") + scale_color_manual(values = Treatment_col) +
  coord_fixed(ratio=1) 
plot_pca_unnorm_by_qc
ggsave("Figs/cleaning/pca_unnorm_by_qc.pdf",plot = plot_pca_unnorm_by_qc)
```



## Check sample validity by sex chromosome gene expression, kinship comparison, and stimulant PCA

```{r inferSexByRna, dependson="applyQcCuts"}
## infer sex based on RNA-seq reads
design_w_QC$logXyRatio <-
  logXYratio(counts.pc %>% relocate(c(geneName,design_w_QC$libid)) %>% column_to_rownames("geneName"), gene_ID = "ensembl",use_annotables = FALSE)
# plot histogram, and use it to determine break point

logXyRatioThreshold <- 6.5
pal.sex <-
  c("F"="orange", "M"="blue")
design_w_QC$sexByRna <-
  ifelse(design_w_QC$logXyRatio >= logXyRatioThreshold, "F", "M") %>%
  factor(levels=c("F", "M"))
```

```{r plotLogXyRatio, dependson="inferSexByRna", fig.width=7.25, fig.height=4.25}
plot.tmp <-
  ggplot(
    design_w_QC,
    mapping = aes(x = logXyRatio, fill = Sex)) +
  geom_histogram(color = "black", position = "dodge") +
  geom_vline(xintercept = logXyRatioThreshold, linetype = "dashed") +
  labs(x = "log ratio of X to Y chromosome reads") +
  scale_fill_manual("sex\n(reported)", values = pal.sex, na.value="gray60")

print(plot.tmp)

```



None of the samples seem to be have mismatched sex





# 4. Create DGEList object

```{r dgelist}
dat.pc <- DGEList(
  #count table. move gene names to column names
  counts=as.matrix(column_to_rownames(counts.pc,
                                      "geneName")),
  #metadata
  samples=design_w_QC %>% arrange(libid),
  #keep genes in count table
  genes=key.pc)

save(dat.pc, 
     file=file.path(dirDataclean, paste(projectNumber,"_dat_clean_unnormalized.RData",sep="")))
```

# 5. Rare gene filtering
The raw gene sets contain highly variable, low abundance/rare genes. 



## Threshold for filtering : 
  Present in minimum sample = 8 
  minimum CPM = 0.8

```{r}
load("Data_clean/P513-2_dat_clean_unnormalized.RData")

dat.pc.abund<-RNAetc::filter_rare(dat = dat.pc, min.pct = 10,min.CPM = 0.3,gene.var = "geneName",plot = TRUE)
```



# 6. Normalize the data

## 6.1 Calculate factors to scale library sizes.

```{r norm}
dat.pc.abund.norm <- calcNormFactors(dat.pc.abund)
```






## 6.2 Normalize with voom

```{r voom3}
dat.pc.abund.norm.voom <- voomWithQualityWeights(
                          dat.pc.abund.norm,
                          design=model.matrix(~Treatment,
                                 data=dat.pc.abund.norm$samples),
                          plot=TRUE)

save(dat.pc.abund.norm.voom,file="Data_clean/dat_pc_abund_norm_voom.RData")
```


# 7. Save the data

```{r}

dat.voom <- dat.pc.abund.norm.voom
clean_col_names <- clean_column_names(dat.voom$targets %>% colnames())
setnames(dat.voom$targets, colnames(dat.voom$targets), clean_col_names)
dat.voom$targets <- dat.voom$targets  %>%
  mutate(viral_load = RV16_Copy_Number_Mean) %>% 
  mutate(logviral_load = log2(viral_load),log10viral_load = log10(viral_load))



save(dat.voom, 
     file=file.path(dirDataclean, paste(projectNumber,"_voom_normalized.RData",sep="")))
```

```{r}
dat.voom_batch_corrected_limma_removed_replicates_batch_3 <- readRDS("../P513-3/Data_clean/P513-3_voom_normalized_Limma_batch_corrected_wo_replicates.rds")

meta_w_batch3<- dat.voom_batch_corrected_limma_removed_replicates_batch_3$targets%>%
  subset(timePoint == "48hr"&treatment == "No_Treatment") %>% 
  select(donorId,project) %>% distinct() %>% subset(donorId %in% unique(dat.module$targets$donorId)) %>% rename("projectId"="project")
```

Add to the metadata and make PCA
```{r}
dat.module$targets <- dat.module$targets %>% left_join(meta_w_batch3)
dat.voom$targets <- dat.voom$targets %>% left_join(meta_w_batch3)
```


# 8. PCA final data set




## 8.1 PCA by Treatment
```{r  echo=FALSE, warning=FALSE}

#Calculate PCA for voom norm data.
PCA.voom <- as.data.frame(dat.voom$E) %>% 
  t() %>% 
  #Calc PCA
  prcomp(scale. = TRUE)
PC1.label <- paste("PC1 (", summary(PCA.voom)$importance[2,1]*100, "%)", sep="")
PC2.label <-paste("PC2 (", summary(PCA.voom)$importance[2,2]*100, "%)", sep="")
PC3.label <-paste("PC3 (", summary(PCA.voom)$importance[2,3]*100, "%)", sep="")
# Extract PC values
PCA.voom.dat<- as.data.frame(PCA.voom$x) %>% 
  rownames_to_column("libid") %>%
  # Select PCs for plotting
  dplyr::select(libid, PC1:PC3) %>% 
  # Merge with metadata
  left_join(as.data.frame(dat.voom$targets),
            by="libid")
plot_normalized<-ggplot(PCA.voom.dat ,
                  aes(PC1, PC2)) +
           geom_point(aes(color=Treatment),
                      size=3) +
  #Beautify
  theme_classic() +
  theme(legend.title = element_blank(),
        legend.position = "right",legend.text = element_text(size=12)) + 
  labs(x=PC1.label, y=PC2.label, title="Voom normalized log2 CPM") +scale_color_manual(values = Treatment_col) +
  coord_fixed(ratio=1)
plot_normalized
ggsave("figs/cleaning/PCA_normalized_treatment.pdf")


```


```{r}

#dir.create(file.path(dirPlots,"cleaning/PCA"),recursive = TRUE)
for (col_ in colnames(dat.voom$targets)) {
ggplot(PCA.voom.dat, 
    aes(x = PC1, y = PC2, color = .data[[col_]])) + 
    geom_point(size = 3) + 
    theme_classic() +
    theme(legend.title = element_blank(),
          legend.position = "right", 
          legend.text = element_text(size = 12)) + 
    labs(x = PC1.label, y = PC2.label, title = paste("PCA by", col_, sep = " ")) +
    coord_fixed(ratio = 1)
  ggsave(paste0("figs/cleaning/PCA/", col_, ".pdf"),height = 5,width = 5)
}

variables_of_interest <- c("donorId","median_cv_coverage","fastq_total_reads","stimulation","Treatment","Sex","T2_High_phenotype_elevated_FENO_and_or_eos","log10viral_load","Exacerbation","projectId")


PC1_PC2_PCA_variables_of_interest_list <- list()

for (col_ in variables_of_interest) {
  
  # Check if the column is numeric (continuous) or categorical
  if (is.numeric(PCA.voom.dat[[col_]])) {
    color_scale <- scale_color_viridis_c()  # Continuous: Viridis scale
  } else {
    num_categories <- length(unique(PCA.voom.dat[[col_]]))
    
    # Define color palette dynamically
    color_palette <- if (num_categories <= 8) {
      scale_color_brewer(palette = "Set2")  # Good for categorical variables
    } else {
      scale_color_manual(values = rainbow(num_categories))  # Fallback for many categories
    }
    
    color_scale <- color_palette
  }

  # Generate the PCA plot
  PC1_PC2_PCA_variables_of_interest_list[[col_]] <- ggplot(PCA.voom.dat, 
    aes(x = PC1, y = PC2, color = .data[[col_]])) + 
    geom_point(size = 3) + 
    color_scale +  # Apply selected color scale
    theme_classic() +
    theme(legend.title = element_blank(),
          legend.position = "right", 
          legend.text = element_text(size = 12)) + 
    labs(x = PC1.label, y = PC2.label, title = paste("PCA by", col_, sep = " ")) +
    coord_fixed(ratio = 1)
}

PCA_PC1_PC2_var_of_int <- patchwork::wrap_plots(PC1_PC2_PCA_variables_of_interest_list, ncol = 3)
ggsave(file.path(dirPlots,"cleaning/PCA_PC1_PC2_variable_of_int.pdf"),plot = PCA_PC1_PC2_var_of_int,height = 10,width = 15)


PC1_PC3_PCA_variables_of_interest_list <- list()

for (col_ in variables_of_interest) {
  
  # Check if the column is numeric (continuous) or categorical
  if (is.numeric(PCA.voom.dat[[col_]])) {
    color_scale <- scale_color_viridis_c()  # Continuous: Viridis scale
  } else {
    num_categories <- length(unique(PCA.voom.dat[[col_]]))
    
    # Define color palette dynamically
    color_palette <- if (num_categories <= 8) {
      scale_color_brewer(palette = "Set2")  # Good for categorical variables
    } else {
      scale_color_manual(values = rainbow(num_categories))  # Fallback for many categories
    }
    
    color_scale <- color_palette
  }

  # Generate the PCA plot
  PC1_PC3_PCA_variables_of_interest_list[[col_]] <- ggplot(PCA.voom.dat, 
    aes(x = PC1, y = PC3, color = .data[[col_]])) + 
    geom_point(size = 3) + 
    color_scale +  # Apply selected color scale
    theme_classic() +
    theme(legend.title = element_blank(),
          legend.position = "right", 
          legend.text = element_text(size = 12)) + 
    labs(x = PC1.label, y = PC3.label, title = paste("PCA by", col_, sep = " ")) #+
    #coord_fixed(ratio = 1)
}
PCA_PC1_PC3_var_of_int <- patchwork::wrap_plots(PC1_PC3_PCA_variables_of_interest_list, ncol = 3)
ggsave(file.path(dirPlots,"cleaning/PCA_PC1_PC3_variable_of_int.pdf"),plot = PCA_PC1_PC3_var_of_int,height = 10,width = 15)

```




# 13. Co-variate correlation with PC



```{r, fig.width=8.5,fig.height=8.5, message=FALSE}
#PCA percents
pca <- stats::prcomp(t(dat.voom$E), scale.=TRUE)
pct <- summary(pca)$importance %>% t() %>% 
  as.data.frame() %>% 
  rownames_to_column("name") %>% 
  mutate(newname = paste(name, " (", signif(`Proportion of Variance`*100, 2), 
                      "%)", sep="")) %>% 
  select(name, newname)
#save(pct, file = "data_clean/PC_labs_voom_normalized.RData")
#PC values
pca.dat <- as.data.frame(pca$x) %>% 
  rownames_to_column() %>% 
  pivot_longer(-rowname) %>% 
  left_join(pct, by = "name") %>% 
  select(-name) %>% 
  pivot_wider(names_from = newname) %>% 
  arrange(rowname) %>% 
  column_to_rownames() %>%  select(1:13)

#Co-variates as numeric
meta.dat.cov <- dat.voom$targets %>% 
  arrange(libid) %>%
  select(donorId, median_cv_coverage,fastq_total_reads,treatment,stimulation,Treatment,
          Age_yrs,Sex,libid,T2_High_phenotype_elevated_FENO_and_or_eos,Exacerbation,Race_Ethnicity,FEV1_predicted,
         FEV1_FVC_ratio,FEF25_75_predicted,Inhaled_corticosteroid_ICS_use_at_time_of_AEC_collection,log10viral_load,projectId
         ) %>% 
  mutate_if(is.character, ~as.numeric(as.factor(.))) %>% 
  mutate_if(is.factor, ~as.numeric(.)) %>% 
  as.matrix()
rownames(meta.dat.cov) <- sort(dat.voom$targets$libid)

#Correlation
corr <- psych::corr.test(pca.dat, meta.dat.cov, method = "kendall", adjust = "none")

#heatmap
corrplot(t(corr$r), p.mat=t(corr$p), sig.level = c(0.01, 0.05), pch.cex = 0.9,
         insig = 'label_sig', method = "color", tl.col = 'black',
         col = rev(COL2('RdBu', 10)), cl.ratio = 0.2, 
         title = "Kendall correlation (r)\n **P < 0.01 *P < 0.05",
         mar=c(0,0,2,0))
```


```{r include=FALSE}
#save
 
pdf(file.path(dirPlots,"cleaning/PCA_correlation_heatmap_voom_normalized.pdf"), height=5, width=10)
corrplot(t(corr$r), p.mat=t(corr$p), sig.level = c(0.01, 0.05), pch.cex = 0.9,
         insig = 'label_sig', method = "color", tl.col = 'black',
         col = rev(COL2('RdBu', 10)), cl.ratio = 0.2, 
         title = "Kendall correlation (r)\n **P < 0.01 *P < 0.05",
         mar=c(0,0,2,0))
dev.off()
```

# Make the modules from P513 to this data
Make Module data from Batch 1 
```{r}

load("/Users/bbenson/Library/CloudStorage/Box-Box/Altman_Lab/P513_1/Results/Module_obj_min_pct_10.RData")
mods_from_P513 <-module_obj_min_pct_10$mods %>%
      #Combine count and module data
      dplyr::select(geneName, module.char) %>%
      dplyr::left_join(tibble::rownames_to_column(as.data.frame(dat.voom$E), "geneName"),
                       by="geneName") %>%
      #Calculate mean by module
      dplyr::group_by(module.char) %>%
      dplyr::summarise_if(is.numeric, mean, na.rm = TRUE) %>%
      tibble::rownames_to_column() %>%
      #Make module names
      dplyr::mutate(rowname=paste("module", module.char, sep="_")) %>%
      tibble::column_to_rownames() %>% as.data.frame() %>% select(-c(module.char)) %>% rownames_to_column("Module")

write_csv(mods_from_P513,file.path(dirDataclean,"Module_counts_from_P513.csv"))

write_csv(module_obj_min_pct_10$mods,"Data_clean/Genes_in_Modules_P513-1(Manuscript).csv")
```

# Make Elist for module
```{r dgelist}

dat.module <- new("EList")
dat.module$E<-as.matrix(mods_from_P513 %>% column_to_rownames("Module") %>% select(dat.voom$targets %>% arrange(libid) %>% pull(libid)))
dat.module$targets <- dat.voom$targets %>% arrange(libid)
dat.module$modules <- module_obj_min_pct_10$mods

save(dat.module, 
     file=file.path(dirDataclean,paste(projectNumber,"_Module.RData",sep = "")))
```



# Look at module coherance
```{r}
source("../../Bioinformatics_R_scripts/calculate_module_coherence.R")
P513_4_mods_batch1_coherence <- 
calculate_module_coherence(mods = module_obj_min_pct_10$mods,
 dat = dat.voom, mods_title = "P513-1 Modules",
 dat_title = "P513-4", r_cutoff = 0.3, p_cutoff = 0.05)
```



# R session

```{r}
sessionInfo()
```

